platform/darwin/src/local_glyph_rasterizer.mm

```diff
@@ -45,47 +46,72 @@ CTFontDescriptor configuration option (although we'd have to override font size
-            return NULL;
+    bool isEnabled() { return fallbackFontNames; }
+    
+    CTFontDescriptorRef createFontDescriptor(const FontStack& fontStack) {
+        NSMutableArray *fontNames = [NSMutableArray arrayWithCapacity:fontStack.size() + fallbackFontNames.count];
+        for (auto& fontName : fontStack) {
+            if (fontName != util::LAST_RESORT_ALPHABETIC_FONT && fontName != util::LAST_RESORT_PAN_UNICODE_FONT) {
+                [fontNames addObject:@(fontName.c_str())];
+            }
         }
+        [fontNames addObjectsFromArray:fallbackFontNames];
         
-        if (!fontHandle) {
-          NSDictionary *fontAttributes = @{
-                (NSString *)kCTFontSizeAttribute: [NSNumber numberWithFloat:24.0],
-                (NSString *)kCTFontFamilyNameAttribute: [[NSString alloc] initWithCString:fontFamily->c_str() encoding:NSUTF8StringEncoding]
+        CFMutableArrayRefHandle fontDescriptors(CFArrayCreateMutable(kCFAllocatorDefault, fontNames.count, &kCFTypeArrayCallBacks));
+        for (NSString *name in fontNames) {
+            NSDictionary *fontAttributes = @{
+                (NSString *)kCTFontSizeAttribute: @(util::ONE_EM),
+                (NSString *)kCTFontNameAttribute: name,
+                (NSString *)kCTFontDisplayNameAttribute: name,
+                (NSString *)kCTFontFamilyNameAttribute: name,
             };
-
+            
             CTFontDescriptorRefHandle descriptor(CTFontDescriptorCreateWithAttributes((CFDictionaryRef)fontAttributes));
-            fontHandle = CTFontCreateWithFontDescriptor(*descriptor, 0.0, NULL);
-            if (!fontHandle) {
-                throw std::runtime_error("CTFontCreateWithFontDescriptor failed");
-            }
+            CFArrayAppendValue(*fontDescriptors, *descriptor);
+        }
+
+        CFStringRef keys[] = { kCTFontSizeAttribute,                  kCTFontCascadeListAttribute };
+        CFTypeRef values[] = { (__bridge CFNumberRef)@(util::ONE_EM), *fontDescriptors };
+
+        CFDictionaryRefHandle attributes(
+            CFDictionaryCreate(kCFAllocatorDefault, (const void**)&keys,
+                (const void**)&values, sizeof(keys) / sizeof(keys[0]),
+                &kCFTypeDictionaryKeyCallBacks,
+                &kCFTypeDictionaryValueCallBacks));
+        if (CFArrayGetCount(*fontDescriptors)) {
+            CTFontDescriptorRef firstDescriptor = (CTFontDescriptorRef)CFArrayGetValueAtIndex(*fontDescriptors, 0);
+            return CTFontDescriptorCreateCopyWithAttributes(firstDescriptor, *attributes);
+        } else {
+            return CTFontDescriptorCreateWithAttributes(*attributes);
         }
-        return fontHandle;
+    }
+
+    CTFontRef createFont(const FontStack& fontStack) {
+        CTFontDescriptorRefHandle descriptor(createFontDescriptor(fontStack));
+        return CTFontCreateWithFontDescriptor(*descriptor, 0.0, NULL);
     }
     
 private:
-    optional<std::string> fontFamily;
-    CTFontRef fontHandle;
```

Comment:
This PR no longer memoizes the resolved font. The font cascade list starts with the font stack from the style, which means the font descriptor can potentially differ for every invocation of `LocalGlyphRasterizer::rasterizeGlyph()`. If performance turns out to be an issue, this class could build up a cache mapping `FontStack`s to `CTFontDescriptorRef`s.

Reply:
Yep, let's check the performance after this PR.